### PR TITLE
Bug 934608 - npm-mirror doesn't always download all versions when it needs multiple

### PR DESCRIPTION
**Pull request by lambdabaa**

### CODE REVIEW

File: lib/syncmanager.js [hunk 1]
```diff
@@ -322,56 +322,53 @@ SyncManager.prototype = {
...
-          return process();
-        }
-
-        var packageVersionUrl = Package.url(this.master, package, version);
-        Download.inst.download(
-          packageVersionUrl, function(e, packageVersionData) {
-          if (e) {
-            debug('error downloading package ' + package + ': ' + e);
+      var versions = Object.keys(this.packageToVersions[package]);
+      versions.forEach(function(version) {
```
Comment (lambdabaa) [inline lib/syncmanager.js:326]: The important change is here. Before I had a for in loop and I was running into the interview question problem :)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "npm-mirror",
-  "version": "0.0.1-alpha-2",
+  "version": "0.1.0-alpha",
```
Comment (lightsofapollo) [inline package.json:3]: =p is this really alpha

File: lib/syncmanager.js [hunk 3]
```diff
@@ -322,56 +322,53 @@ SyncManager.prototype = {
...
+            copy.dependencies = packageVersion.dependencies;
+            copy.devDependencies = packageVersion.devDependencies;
+            copy.peerDependencies = packageVersion.peerDependencies;
+            copy.dist = {};
+            copy.dist.tarball =
+              Package.tarballUrl(this.master, package, version);
+
+            fs.writeFile(dest, JSON.stringify(copy), function() {
+              var tarball = packageVersion.dist.tarball;
+              var dest = path.resolve(
```
Comment (lightsofapollo) [inline lib/syncmanager.js:358]: ( ) on one line

Comment (lightsofapollo) [inline lib/syncmanager.js:358]: ( ) on their own lines if you need to wrap:

``` js
do(
  magic,
  too
);
```

File: package.json (hunk 2 above)
Comment (lambdabaa) [inline package.json:3]: hahaha

**Pull request merged**